### PR TITLE
 Make cumulative sprop a typing flag, deprecate command line -sprop-cumulative

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

File: toplevel/coqargs.ml [hunk 1]
```diff
@@ -523,7 +527,9 @@ let parse_args ~help ~init arglist : t * string list =
       set_logic (fun o -> { o with impredicative_set = Declarations.ImpredicativeSet }) oval
     |"-allow-sprop" -> set_logic (fun o -> { o with allow_sprop = true }) oval
     |"-disallow-sprop" -> set_logic (fun o -> { o with allow_sprop = false }) oval
-    |"-sprop-cumulative" -> set_logic (fun o -> { o with cumulative_sprop = true }) oval
+    |"-sprop-cumulative" ->
+      warn_deprecated_sprop_cumul();
+      set_logic (fun o -> { o with cumulative_sprop = true }) oval
```
Comment (Zimmi48) [inline toplevel/coqargs.ml:532]: If #11997 gets merged first, you should use `add_set_option` instead.

**Review by Zimmi48 — COMMENTED** [review]
Looks good to me. To me, the deprecation of the command-line argument is not required, given that we have a few other vernac options which have a corresponding dedicated command-line option (even though these are made non-essential by the generic `-set` / `-unset` command-line options). That being said, it is also OK to decide that this option is not essential enough to deserve a dedicated command-line version.

File: toplevel/coqargs.ml (hunk 1 above)
Comment (Zimmi48) [reply toplevel/coqargs.ml:532]: #11997 has been merged. Please update to use `add_set_option`.

**Review by Zimmi48 — APPROVED** [review]

**Comment by Zimmi48** [comment]
CI is completely broken. What's going on?

**Comment by SkySkimmer** [comment]
coqchk values.ml broken

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
This PR is not really making the situation worse, but I do feel queasy every time I have a look at the way we handle trust in the checker. In particular, we shouldn't liberally accept typing flags known to make the system broken without some opt-in flag or a red blinking warning. See #11477.

**Pull request merged**